src/viewport/index.tsx

```diff
@@ -547,66 +543,36 @@ class Viewport extends React.Component<
         }
     }
 
-    public changeAgentColor(
-        agents: string | SelectionEntry[],
-        color: string
-    ): void {
-        if (typeof agents === "string") {
-            agents = this.convertAgentNameToSelectionEntry(agents);
-        }
-        const agentIds = this.selectionInterface.getColorChangeAgentIds(agents);
-        const colorId = this.getColorId(
-            color,
-            this.props.agentColors as string[]
-        );
-        this.applyColorToAgents(agentIds, colorId);
+    private getColorId(color: string): number {
+        /**
+         * Check if the new color is in our current array of color options, if not,
+         * add it before returning the index
+         */
+        return this.visGeometry.addNewColor(color);
```

Comment:
there was already a function in visGeometry for adding a color to the color data